processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -370,6 +462,21 @@ private static int compareDims(List<DimensionSpec> dimensions, Row lhs, Row rhs)
     return 0;
   }
 
+  private static int compareDimsAndAggs(List<String> fields, List<Integer> directions, Row lhs, Row rhs)
+  {
+    for (int i = 0; i < fields.size(); i++) {
+      final String fieldName = fields.get(i);
+      final int dimCompare = NATURAL_NULLS_FIRST.compare(
+          lhs.getRaw(fieldName),
+          rhs.getRaw(fieldName)
+      );
+      if (dimCompare != 0) {
+        return dimCompare * directions.get(i);
```

Comment:
When limit push down is not used, it will just sort by the default order (all dims, no aggs)